crates/pow/src/lib.rs

```diff
@@ -2,32 +2,51 @@ use std::num::NonZeroUsize;
 use std::sync::{Arc, Mutex};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-use miden_client::account::AccountId;
 use tokio::time::Duration;
 
-use crate::pow::api_key::ApiKey;
-use crate::pow::challenge::Challenge;
-use crate::pow::challenge_cache::ChallengeCache;
+use crate::challenge_cache::ChallengeCache;
 
-pub mod api_key;
-pub mod challenge;
+mod challenge;
 mod challenge_cache;
 
+pub use challenge::Challenge;
+
 // POW
 // ================================================================================================
 
+/// Proof-of-Work implementation.
+///
+/// This struct is used to generate and validate `PoW` challenges.
 #[derive(Clone)]
-pub(crate) struct PoW {
+pub struct PoW {
```

Comment:
I wonder if we should rename this struct (and update the associated docs) into `PoWRateLimiter` (or maybe just `RateLimiter`).